zekrom-vale/ZekDex
# Enhancements to Dataset Documentation and Wiki Generation for ZekDex

## Pull Request for ZekDex

**Issue Being Addressed**
This PR addresses several issues:
- Fixes #9 by ensuring CSV will quote everything
- Works on #5 and #7 by creating a wiki update action, moving data to the wiki, migrating dataset docs to yaml, expanding on dataset descriptions, fixing the issue of missing language yaml, generating mans, making README.md generated by gen_readme.R, generating the wiki files with gen_md.R, generating the sidebar by directory, changing it to generate md, and updating it to a different format

**Summary of Changes**
The changes include a series of commits that primarily focus on improving the dataset documentation and the generation of markdown files for the wiki. This includes the creation of a wiki update action, moving data to the wiki, migrating dataset docs to yaml, expanding on dataset descriptions, fixing the issue of missing language yaml, generating mans, making README.md generated by gen_readme.R, generating the wiki files with gen_md.R, generating the sidebar by directory, changing it to generate md, updating it to a different format, ensuring CSV will quote everything, adding a ggplot fill based on type, fixing rendering of defending_*, regenerating data, adding missed exports, ignoring some files, and incrementing the version.

**Update Data or @importFrom Directives**
The changes involved regenerating data. If new data was added, it can be updated in the future by following the same process used in this PR.

**Testing Instructions**
```r
devtools::check()
0 errors ✔ | 7 warnings ✖ | 3 notes ✖
Error: R CMD check found WARNINGs
Execution halted
```
Please note that the check resulted in warnings and notes. These will be addressed later.

**Additional Notes**
This PR includes a series of changes that significantly improve the documentation and organization of the ZekDex package. It addresses several issues and makes the package easier to use and understand. However, the `devtools::check()` resulted in some warnings and notes that need to be addressed.
